DQMServices/Core/interface/locking_ptr.h

```diff
@@ -0,0 +1,136 @@
+#ifndef locking_ptr_h
+#define locking_ptr_h
+
+#include <memory>
+
+namespace detail {
+
+  class lock_base
+  {
+  public:
+    lock_base() = default;
+    virtual ~lock_base() = default;
+
+    virtual void lock() const = 0;
+    virtual void unlock() const = 0;
+  };
+
+  template <typename L>
+  class lock_impl : public lock_base
```

Comment:
Why bother creating a separate and **our own** definition of what a lock is? Why not simply a use a concrete mutex as a template parameter to template <typename L>.